test/batchReceiver.spec.ts

```diff
@@ -234,31 +235,32 @@ describe("ReceiveBatch from Queue/Subscription", function(): void {
 
     await testPeekMsgsLength(queueClient, 0);
 
-    const deadLetterQueuePath = process.env.QUEUE_NAME + "/$DeadLetterQueue";
-    const client = namespace.createQueueClient(deadLetterQueuePath);
-    receivedMsgs = await client.receiveBatch(1);
+    const deadLetterQueuePath = Namespace.getDeadLetterQueuePathForQueue(
+      process.env.QUEUE_NAME ? process.env.QUEUE_NAME : ""
+    );
+    const deadletterQueueClient = namespace.createQueueClient(deadLetterQueuePath);
+    const deadLetterMsgs = await deadletterQueueClient.receiveBatch(1);
 
-    should.equal(Array.isArray(receivedMsgs), true);
-    should.equal(receivedMsgs.length, 1);
-    should.equal(receivedMsgs[0].body, testMessages[0].body);
-    should.equal(receivedMsgs[0].messageId, testMessages[0].messageId);
+    should.equal(Array.isArray(deadLetterMsgs), true);
+    should.equal(deadLetterMsgs.length, 1);
+    should.equal(deadLetterMsgs[0].body, testMessages[0].body);
+    should.equal(deadLetterMsgs[0].messageId, testMessages[0].messageId);
 
-    await receivedMsgs[0].complete();
+    await deadLetterMsgs[0].complete();
 
-    await testPeekMsgsLength(client, 0);
-    await client.close();
+    await testPeekMsgsLength(deadletterQueueClient, 0);
+    await deadletterQueueClient.close();
   });
 
-  it("Abandoning a message 10 times, you get it no more. You can then only get it from the dead letter subscription", async function(): Promise<
+  it("Message abandoned more than maxDeliveryCount goes to dead letter subscriptions", async function(): Promise<
     void
   > {
     await topicClient.send(testMessages[0]);
     let abandonMsgCount = 0;
-    let receivedMsgs;
-    let clientSubscription: SubscriptionClient;
+    let deadletterSubscriptionClient: SubscriptionClient;
```

Comment:
Do we have to declare this here? Can we not declare it as a constant when we create the queue client?

Reply:
Did you mean globally? outside of the method?

Reply:
No no :) Just move it to where it is actually being used